### PR TITLE
Small changes for chapter 0

### PR DESCRIPTION
**Pull request by MithicSpirit**
Most changes here are also discussed in @MAO3J1m0Op's review. Some of
the more notable changes are:
- using \`foo\` instead of "foo" when referring to code and such
- using logical quotes instead traditional quotes (i.e., "foo". instead
  of "foo.")
- improve comment formatting
- improve explanation of `True` and `False`
- improve explanation of `exfalso`
- use `nth_rw` instead of `nth_rewrite`

### CODE REVIEW

File: AlgebraInLean/World0LeanIntroSOLUTIONS/Sheet2Solutions.lean [hunk 1]
```diff
@@ -31,75 +33,79 @@ example:
...
-/-Let's move on beyond the NNG to do some exercises you haven't seen before. The following exercises
-are from Kevin Buzzard's Formalizing Mathematics.-/
+/-
+Let's move on beyond the NNG to do some exercises you haven't seen before. The following exercises
+are from Kevin Buzzard's Formalizing Mathematics.
+-/
 
 variable (P Q R S T : Prop)
-/-Note that the variables we are working with here are of the type Prop.-/
+/- Note that the variables we are working with here are of the type Prop. -/
```
Comment (MAO3J1m0Op) [inline AlgebraInLean/World0LeanIntroSOLUTIONS/Sheet2Solutions.lean:78]: Change these to one-line (`--`) comments?

File: AlgebraInLean/World0LeanIntroSOLUTIONS/Sheet4Solutions.lean [hunk 2]
```diff
@@ -4,42 +4,50 @@ This is a solutions sheet.
...
-With the "constructor" tactic comes the need to talk about some helpful formatting in Lean. The ·
+With the `constructor` tactic comes the need to talk about some helpful formatting in Lean. The `·`
 that you see above is used to indicate steps to solve separate goals.
-⊢ P is one goal, so a new dot is added. exact h1 then solves that goal.
-⊢ Q is the next goal, so another dot is added. exact h2 then solves that goal.
-In the case where more than one tactic is requred to resolve the goal, lines following the line with
-the new · are indented to indicate that they are associated with a certain goal.
 
-For example:-/
+- `⊢ P` is one goal, so a `·` is added. `exact h1` this first goal.
```
Comment (MAO3J1m0Op) [inline AlgebraInLean/World0LeanIntroSOLUTIONS/Sheet4Solutions.lean]: re-add "then solves"

**Review by MAO3J1m0Op — CHANGES_REQUESTED** [review]
fix typo, then you're good

**Review by Stavan-Jain — APPROVED** [review]
lgtm

**Review by chenne19 — APPROVED** [review]

File: AlgebraInLean/World0LeanIntroSOLUTIONS/Sheet2Solutions.lean (hunk 1 above)
Comment (MithicSpirit) [reply AlgebraInLean/World0LeanIntroSOLUTIONS/Sheet2Solutions.lean:78]: Why single-line? Although I do think that the comment should overall be changed to better explain what `variable` does.

**Review by MAO3J1m0Op — APPROVED** [review]
lgtm

**Review by Anoushka1406 — APPROVED** [review]
lgtm

**Pull request merged**